app/controllers/goals_controller.rb

```diff
@@ -12,6 +12,22 @@ def create
     end
   end
 
+  def destroy
```

Comment:
> does a more specific query improve database efficiency?

Usually, though to be sure we'd need to benchmark the query times and compare them.

However, the potential performance gain isn't the reason for the current_user-scoped query. Its that it will return nil if the logged in user queries for a goal id that is for a goal belonging to another user. The current_user-scoped query provides a layer of security that isn't there with the alternative. We can walkthrough some demonstrations of this in the call - bring it up if you're interested.